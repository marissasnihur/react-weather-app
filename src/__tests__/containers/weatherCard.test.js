import React from 'react';
import ReactDOM from 'react-dom';

import generateForecasts from '../../containers/helpers/generateForecast';

import WeatherCard from '../../containers/weatherCard';
import CardTemperature from  '../../components/temperature/temperature';
import CardDescription from  '../../components/currentCondition/currentCondition';
import CardTitle from  '../../components/cityNameTitle/cityNameTitle';

const mockForecastData = [{
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}];

describe('WeatherCard container...', () => {
  const staticWeatherCard = (
    <WeatherCard
      city="somewhere"
      temperature="30"
      currentCondition="clear"
      cityImage="http://example.com/image.jpg"
      forecast={[]}
    />);

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCard, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCard, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('WeatherCard bottom portion...', () => {
  const staticWeatherCardBottom =
    <div className='cardBottomPortion'>
      <div className='cardDetail'>
        <ul className='forecastsWrapper'>
          {generateForecasts(mockForecastData)}
        </ul>
      </div>
    </div>;

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('WeatherCard top component...', () => {
  const staticWeatherCardBottom =
    <div className='cardUpperPortion' style={{"backgroundImage": 'url("/assets/images/testimage.jpg")'}}>
      <div className='cardInfo'>
        <CardTemperature>10</CardTemperature>
        <CardDescription>Clear</CardDescription>
      </div>
      <CardTitle>Somewhere</CardTitle>
    </div>;

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should create the background image inline style correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    expect(div.firstChild.getAttribute('style')).toBe('background-image: url(/assets/images/testimage.jpg);');
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});