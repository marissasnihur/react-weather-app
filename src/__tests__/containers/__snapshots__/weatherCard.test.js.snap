// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard bottom portion... should match snapshot 1`] = `
<div>
  <div
    class="cardBottomPortion"
  >
    <div
      class="cardDetail"
    >
      <ul
        class="forecastsWrapper"
      >
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Sat
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Sun
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Mon
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Tue
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Wed
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`WeatherCard container... should match snapshot 1`] = `
<div>
  <div
    class="card "
  >
    <div
      class="cardUpperPortion"
      style="background-image: url(http://example.com/image.jpg);"
    >
      <div
        class="cardInfo"
      >
        <div
          class="cardTemperature"
        >
          30
        </div>
        <div
          class="currentCondition"
        >
          clear
        </div>
      </div>
      <h2
        class="cityNameTitle"
      >
        somewhere
      </h2>
    </div>
    <div
      class="cardBottomPortion"
    >
      <div
        class="cardDetail"
      >
        <ul
          class="forecastsWrapper"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherCard top component... should match snapshot 1`] = `
<div>
  <div
    class="cardUpperPortion"
    style="background-image: url(/assets/images/testimage.jpg);"
  >
    <div
      class="cardInfo"
    >
      <div
        class="cardTemperature"
      >
        10
      </div>
      <div
        class="currentCondition"
      >
        Clear
      </div>
    </div>
    <h2
      class="cityNameTitle"
    >
      Somewhere
    </h2>
  </div>
</div>
`;
